File: Resource/51900286-51900712-51900718-51900699/source demo/Window Form/51900286-51900712-51900718-51900699/RestaurentDTO/ChiTietGoiMon.cs
using System.Data;

namespace RestaurentDTO
{
    public class ChiTietGoiMon
    {
        private string _maMonAn;
        private string _maOrder;
        private int _soLuongMonAn;

        public ChiTietGoiMon(string maMonAn, string maOrder, int soLuongMonAn)
        {
            _maMonAn = maMonAn;
            _maOrder = maOrder;
            _soLuongMonAn = soLuongMonAn;
        }

        public ChiTietGoiMon(DataRow dataRow)
        {
            _maMonAn = dataRow["MaMonAn"].ToString();
            _maOrder = dataRow["MaOrder"].ToString();
            _soLuongMonAn = (int)dataRow["SoLuongMonAn"];
        }

        public string MaMonAn { get => _maMonAn; set => _maMonAn = value; }
        public string MaOrder { get => _maOrder; set => _maOrder = value; }
        public int SoLuongMonAn { get => _soLuongMonAn; set => _soLuongMonAn = value; }
    }
}

File: Dev/Models/Models/ChiTietGoiMon.java
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
@Table(name = "chitietgoimon")
public class ChiTietGoiMon  {
    @EmbeddedId
    ChitietgoimonKey id;

    @ManyToOne
    @MapsId("MaMonAn")
    @JoinColumn(name = "MaMonAn")
    Monan monan;


    @ManyToOne
    @MapsId("MaOrder")
    @JoinColumn(name = "MaOrder")
    Phieugoimon phieugoimon;

    private int SoLuongMonAn;
}